src/AppInstallerCommonCore/AppInstallerTelemetry.cpp

```diff
@@ -91,16 +109,54 @@ namespace AppInstaller::Logging
         m_isRuntimeEnabled = true;
     }
 
+    void TelemetryTraceLogger::SetTelemetryCorelationJson(const std::wstring_view& jsonStr_view) noexcept
+    {
+        // Check if passed in string is a valid Json formatted before returning the value
+        // If invalid, return empty Json
+        Json::CharReaderBuilder jsonBuilder;
+        Json::CharReader* jsonReader = jsonBuilder.newCharReader();
+        std::unique_ptr<Json::Value> pJsonValue = std::make_unique<Json::Value>();
+        std::string errors;
+        std::wstring jsonStrW{ jsonStr_view };
+        std::string jsonStr = ConvertToUTF8(jsonStrW.c_str());
+
+        bool result = jsonReader->parse(jsonStr.c_str(),
+            jsonStr.c_str() + jsonStr.size(),
+            pJsonValue.get(),
+            &errors);
+
+        delete jsonReader;
```

Comment:
This should be a `std::unique_ptr`

Reply:
Done.

Reply:
Json::CharReader has virtual methods and is abstract class which cannot be instantiated in make_unique. The derived class OurCharReader isn't abstract whose object pointer newCharReader() is actually returning but it is not declared in Json.h to make unique_ptr to it. How to use std::unique_ptr here ?

Reply:
Right. Thanks.